### PR TITLE
Rocky8 for gpdb6

### PR DESCRIPTION
**Pull request by lucasbonner**
[GPR-1083]

Authored-by: Lucas Bonner <blucas@vmware.com>

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by khuddlefish — APPROVED** [review]

File: concourse/pipelines/gen_pipeline.py [hunk 1]
```diff
@@ -297,7 +297,7 @@ def main():
         action='store',
         dest='os_types',
         default=['centos6'],
-        choices=['centos6', 'centos7', 'rhel8', 'oracle7', 'photon3', 'sles12', 'ubuntu18.04', 'win'],
+        choices=['centos6', 'centos7', 'rhel8', 'oracle7', 'photon3', 'sles12', 'ubuntu18.04', 'rocky8', 'win'],
         nargs='+',
```
Comment (BaiShaoqi) [inline concourse/pipelines/gen_pipeline.py:301]: the diffs for the gpdb_6X_STABLE-generated.yml is not included in the PR.
If we are not going to make the pipeline changes now to add rocky8 in 6x-stable pipeline. we can skip the changes for gen_pipeline.py, or else if others fly the 6x-stable pipeiline, the rocky8 jobs will be in 6x-stable pipeline.
if we are going to make the pipeline changes now to add rocky8 in 6x-stable pipeline, then also need to commit the changes for gpdb_6X_STABLE-generated.yml

Comment (lucasbonner) [reply concourse/pipelines/gen_pipeline.py]: Removed `gen_pipeline.py` changes. Thanks Shaoqi!

File: concourse/scripts/common.bash [hunk 2]
```diff
@@ -34,7 +34,7 @@ function os_version() {
 function build_arch() {
 	local id=$(os_id)
 	local version=$(os_version)
-	# BLD_ARCH expects rhel{6,7,8}_x86_64 || photon3_x86_64 || sles12_x86_64 || ubuntu18. || rocky8_x86_64
```
Comment (BaiShaoqi) [inline concourse/scripts/common.bash:37]: not sure if there is  a reason why we changed the mode of the file form 755 to 644 ?

File: concourse/pipelines/.gitignore [hunk 3]
```diff
@@ -3,3 +3,4 @@ gpdb-*.yml
 # negate the match on gpdb-tpl.yml so that tools like `ag` don't ignore it
 # when searching for matches
 !gpdb-tpl.yml
+!gpdb-tpl-rocky8
```
Comment (BaiShaoqi) [inline concourse/pipelines/.gitignore:6]: i don't think we need this change, since there won't be  rocky8 template file.

**Pull request merged**